osu.Game.Rulesets.Taiko/Mods/TaikoInputBlockingMod.cs

```diff
@@ -0,0 +1,138 @@
+        public override ModType Type => ModType.Conversion;
+
+        private const double flash_duration = 1000;
+
+        private DrawableTaikoRuleset ruleset = null!;
+
+        private TaikoPlayfield playfield { get; set; } = null!;
+
+        protected TaikoAction? LastAcceptedDonAction { get; private set; }
+        protected TaikoAction? LastAcceptedKatAction { get; private set; }
+
+        /// <summary>
+        /// A tracker for periods where alternate should not be forced (i.e. non-gameplay periods).
+        /// </summary>
+        /// <remarks>
+        /// This is different from <see cref="Player.IsBreakTime"/> in that the periods here end strictly at the first object after the break, rather than the break's end time.
+        /// </remarks>
+        private PeriodTracker nonGameplayPeriods = null!;
+
+        private IFrameStableClock gameplayClock = null!;
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<TaikoHitObject> drawableRuleset)
+        {
+            ruleset = (DrawableTaikoRuleset)drawableRuleset;
+            ruleset.InputManager.Add(new InputInterceptor(this));
+            playfield = (TaikoPlayfield)ruleset.Playfield;
+
+            var periods = new List<Period>();
+
+            if (drawableRuleset.Objects.Any())
+            {
+                periods.Add(new Period(int.MinValue, getValidJudgementTime(ruleset.Objects.First()) - 1));
+
+                foreach (BreakPeriod b in drawableRuleset.Beatmap.Breaks)
+                    periods.Add(new Period(b.StartTime, getValidJudgementTime(ruleset.Objects.First(h => h.StartTime >= b.EndTime)) - 1));
+
+                static double getValidJudgementTime(HitObject hitObject) => hitObject.StartTime - hitObject.HitWindows.WindowFor(HitResult.Meh);
+            }
+
+            nonGameplayPeriods = new PeriodTracker(periods);
+
+            gameplayClock = drawableRuleset.FrameStableClock;
+        }
+
+        public void Update(Playfield playfield)
+        {
+            if (nonGameplayPeriods.IsInAny(gameplayClock.CurrentTime))
+            {
+                if (LastAcceptedDonAction != null)
+                    LastAcceptedDonAction = null;
+
+                if (LastAcceptedKatAction != null)
+                    LastAcceptedKatAction = null;
+            }
+        }
+
+        protected abstract bool CheckValidNewAction(TaikoAction action);
+
+        private bool checkCorrectAction(TaikoAction action)
+        {
+            if (nonGameplayPeriods.IsInAny(gameplayClock.CurrentTime))
+                return true;
+
+            switch (action)
+            {
+                case TaikoAction.LeftCentre:
+                case TaikoAction.RightCentre:
+                case TaikoAction.LeftRim:
+                case TaikoAction.RightRim:
+                    break;
+
+                // Any action which is not left or right button should be ignored.
+                default:
+                    return true;
```

Comment:
Not all that familiar with taiko, but is the number of cases that arent these cases larger or smaller? , if there's less of them then the default should likely be these, and have the exceptions listed.

Reply:
Sorry could you rephrase that? I'm not sure what you mean

Reply:
> Sorry could you rephrase that? I'm not sure what you mean

of the actions not listed there is there any? since i cant see any other ones in the `taikoaction` enum, so im not sure this safeguard is neccesary

Reply:
It seems that this is a leftover from Stable to make sure that you can still use smoke
I'll remove it